classes/stock/StockAvailable.php

```diff
@@ -524,6 +524,13 @@ public function postSave()
         );
         $this->setQuantity($this->id_product, 0, $total_quantity, $id_shop, false);
 
+        // ensure product date update after a stock movement
+        Db::getInstance()->execute('UPDATE ' . _DB_PREFIX_ . 'product
+                                    SET `date_upd` = "' . pSQL(date('Y-m-d H:i:s')) . '" WHERE id_product = ' . (int) $this->id_product);
+        Db::getInstance()->execute('UPDATE ' . _DB_PREFIX_ . 'product_shop
+                                    SET `date_upd` = "' . pSQL(date('Y-m-d H:i:s')) . '" WHERE id_product = ' . (int) $this->id_product . '
+                                        AND id_shop = ' . (int) $this->id_shop);
+
```

Comment:
I am not convinced by this fix :thinking: 

Searching a bit more where this
```
    /**
     * Upgrades total_quantity_available after having saved.
     *
     * @see StockAvailableCore::update()
     * @see StockAvailableCore::add()
     */
    public function StockAvailableCore::postSave(
```
is called, I found
```
    /**
     * Upgrades total_quantity_available after having saved.
     *
     * @see ObjectModel::add()
     */
    public function StockAvailableCore::add(
```
,
```
    /**
     * Upgrades total_quantity_available after having update.
     *
     * @see ObjectModel::update()
     */
    public function StockAvailableCore::update(
```
and
```
    /**
     * Removes a given product from the stock available.
     *
     * @param int $id_product
     * @param int|null $id_product_attribute Optional
     * @param Shop|null $shop Shop id or shop object Optional
     *
     * @return bool
     */
    public static function StockAvailableCore::removeProductFromStockAvailable(
```
Then why the `product` state should be updated if it is only about product in stock data that is updated ?
Without more explanations, I don't agree with the first request.

Concerning the second request, as `ps_product_shop` contains already some stock metadata (as `low_stock_threshold`, `low_stock_alert`, `advanced_stock_management` or `pack_stock_type`), I suppose it would be consistent to update this field :thinking: 

Did you check the workflow to ensure that `$this->id_product` and `$this->id_shop` are defined here ?

What do you think about creating some integration test to validate your use case ? :thinking:

Reply:
Hmmm it looks that `ps_product` contains as well some `*_stock_*` metadata ... Then I don't understand the difference between those tables :sweat_smile:

Reply:
Thanks for explanations @Hlavtox ^^